test/stdexec/algos/adaptors/test_transfer_when_all.cpp

```diff
@@ -44,6 +44,11 @@ TEST_CASE("transfer_when_all simple example", "[adaptors][transfer_when_all]") {
   auto op = ex::connect(std::move(snd1), expect_value_receiver<double>{3.1415});
   ex::start(op);
 }
+TEST_CASE("transfer_when_all with no senders", "[adaptors][transfer_when_all]") {
+  auto snd = ex::transfer_when_all(inline_scheduler{});
+  auto op = ex::connect(std::move(snd), expect_void_receiver{});
```

Comment:
I dunno, but I would expect `when_all()` to be equivalent to `just(tuple{})` instead of `just()`. Can you comment on why you opted for the current semantics?

Reply:
when_all concats the values from all the senders, without wrapping them onto tuples. So an empty list of values seems correct?

Reply:
Thank you for the review. In libunifex, `just` is special-cased for 0 arguments, and ends up calling `set_value()`. But here in stdexec things are more regular, and `just` completes by calling `set_value(tuple{})`, which is the desired end result: it allows a generic program like the following:

```
#include <iostream>
#include <tuple>

#include <stdexec/execution.hpp>

auto print = [](const auto &...args) {
  (std::cout << "[" << ... << args) << "]\n";
};

auto do_test(const auto &...args) {
  auto sender = stdexec::when_all(stdexec::just(args)...);
  auto results = stdexec::this_thread::sync_wait(sender);
  std::apply(print, *results);
}

int main() {
  do_test("a", "b", "c");
  do_test("x");
  do_test();
}
```

This program compiles on the current branch, but if `when_all` added an extra layer of 'tuple', this program would need its own special-case code to work around it.

Note that my current suggestion for `when_all` with no senders isn't implemented in terms of version -- by removing even more special cases, everything just takes care of itself. (Sort of! I did introduce a new `if constexpr` test for the case of 0 senders. But that's only an optimization.)

Reply:
> `just` completes by calling `set_value(tuple{})`

huh? When the `just()` sender completes, it calls `set_value()` (or rather `set_value(rcvr)` since `set_value` always takes a receiver).

Reply:
> when_all concats the values from all the senders, without wrapping them onto tuples. So an empty list of values seems correct?

Oh yeah.
<i>/me pours himself another coffee</i>

Reply:
> huh? When the `just()` sender completes, it calls `set_value()` (or rather `set_value(rcvr)` since `set_value` always takes a receiver).

Right, sorry. I'll just say that adding `tuple{}` here would make the case of zero senders the odd-one-out.